docs/build/guides/dapps/state-archival.mdx

```diff
@@ -0,0 +1,436 @@
+- Application continuity: Ensuring contract instances and Wasm code remain live is essential for uninterrupted dapp operation. It is essential to check for contract availability before attempting to interact with the contract after a long period of inactivity.
+
+## Methods of implementing state archival on the client side
+
+### 1. Extending TTL from the smart contract
+
+This method involves invoking the `extend_ttl()` from your smart contract to extend the TTL of the contract instance and its associated data. This method is useful when you want to keep the data accessible for a longer period. To use this method, your contract must not be archived at the time of calling.
+
+`extend_ttl()` has two important parameters (T,N):
+
+- T is threshold, the current ledger height at which the extension should happen.
+- N is the new ledger height at which the data will expire.
+- Current TTL must be less than T for the extension to happen.
+- If N is less than the current ledger height, the TTL will not be extended and the call will be regarded as a no-op.
+- If N is greater than the current ledger height, the TTL will be extended to N.
+
+Let's see a sample of how we can implement this in a smart contract:
+
+```rust
+#![no_std]
+/// This is a simple contract that just extends TTL for its keys.
+/// It's main purpose is to demonstrate how TTL extension can be tested,
+use soroban_sdk::{contract, contractimpl, contracttype, Env};
+
+#[contracttype]
+pub enum DataKey {
+    MyKey,
+}
+
+#[contract]
+pub struct TtlContract;
+
+#[contractimpl]
+impl TtlContract {
+    /// Creates a contract entry in every kind of storage.
+    pub fn setup(env: Env) {
+        env.storage().persistent().set(&DataKey::MyKey, &0);
+        env.storage().instance().set(&DataKey::MyKey, &1);
+        env.storage().temporary().set(&DataKey::MyKey, &2);
+    }
+
+    /// Extend the persistent entry TTL to 5000 ledgers, when its
+    /// TTL is smaller than 1000 ledgers.
+    pub fn extend_persistent(env: Env) {
+        env.storage()
+            .persistent()
+            .extend_ttl(&DataKey::MyKey, 1000, 5000);
+    }
+
+    /// Extend the instance entry TTL to become at least 10000 ledgers,
+    /// when its TTL is smaller than 2000 ledgers.
+    pub fn extend_instance(env: Env) {
+        env.storage().instance().extend_ttl(2000, 10000);
+    }
+
+    /// Extend the temporary entry TTL to become at least 7000 ledgers,
+    /// when its TTL is smaller than 3000 ledgers.
+    pub fn extend_temporary(env: Env) {
+        env.storage()
+            .temporary()
+            .extend_ttl(&DataKey::MyKey, 3000, 7000);
+    }
+}
+
+mod test;
+```
+
+:::info
+
+The above contract shows how to extend the TTL of a `Persistent`, `Instance`, and `Temporary` data entry. The `extend_ttl()` method is used to extend the TTL of the data entry to a new ledger height.
```

Comment:
Generally speaking, TTL extension should be a normal part of the regular operations that people make. So you wouldn't call extend_temporary on its own, but rather would call it as a normal part of some "dao_mint" or something like that. It'd be good if the examples demonstrated that at least in part.